core/src/main/java/hudson/cli/DisablePluginCommand.java

```diff
@@ -0,0 +1,239 @@
+     * @param indent number of spaces at the beginning.
+     * @param format format as in {@link String#format(String, Object...)}
+     * @param arguments arguments to print as in {@link String#format(String, Object...)}
+     */
+    private void printIndented(int indent, String format, String... arguments) {
+        if (indent == 0) {
+            stdout.format(format + "%n", (Object[]) arguments);
+        } else {
+            String f = "%" + indent + "s" + format + "%n";
+            String[] newArgs = new String[arguments.length + 1];
+            newArgs[0] = " ";
+            int i = 1;
+            for (String arg: arguments) {
+                newArgs[i] = arg;
+                i++;
+            }
+            stdout.format(f, newArgs);
+        }
+    }
+
+    /**
+     * Print the result of a plugin disablement with the indent passed.
+     * @param oneResult the result of the disablement of a plugin.
+     * @param indent the initial indent.
+     */
+    private void printResult(PluginWrapper.PluginDisableResult oneResult, int indent) {
+        PluginWrapper.PluginDisableStatus status = oneResult.getStatus();
+        if (quiet && (PluginWrapper.PluginDisableStatus.DISABLED.equals(status) || PluginWrapper.PluginDisableStatus.ALREADY_DISABLED.equals(status))) {
+            return;
+        }
+
+        printIndented(indent, Messages.DisablePluginCommand_StatusMessage(oneResult.getPlugin(), oneResult.getStatus(), oneResult.getMessage()));
+        if (oneResult.getDependantsDisableStatus().size() > 0) {
+            indent += INDENT_SPACE;
+            for (PluginWrapper.PluginDisableResult oneDependantResult : oneResult.getDependantsDisableStatus()) {
+                printResult(oneDependantResult, indent);
+            }
+        }
+    }
+
+    /**
+     * Restart if at least one plugin was disabled in this process.
+     * @param results the list of results after the disablement of the plugins.
+     */
+    private void restartIfNecessary(List<PluginWrapper.PluginDisableResult> results) throws RestartNotSupportedException {
+        if (restart) {
+            for (PluginWrapper.PluginDisableResult oneResult : results) {
+                if (restartIfNecessary(oneResult)) {
+                    break;
+                }
+            }
+        }
+    }
+
+    /**
+     * Restart if this particular result of the disablement of a plugin and its dependant plugins (depending on the
+     * strategy used) has a plugin disablexd.
+     * @param oneResult the result of a plugin (and its dependants).
+     * @return true if it end up in restarting jenkins.
+     */
+    private boolean restartIfNecessary(PluginWrapper.PluginDisableResult oneResult) throws RestartNotSupportedException {
+        PluginWrapper.PluginDisableStatus status = oneResult.getStatus();
+        if (PluginWrapper.PluginDisableStatus.DISABLED.equals(status)) {
+            Jenkins.get().safeRestart();
```

Comment:
Does this wait until the CLI command finishes?

Reply:
I know that, but does it way until the CLI command finishes?

Reply:
It does Jenkins#safeRestart so no, it doesn't

Reply:
I'm not sure if this ideal then, but the "enable-plugin" command works the same, so let's leave it like that.